File: src/CircleAI.Telephony.Plivo/PlivoOptions.cs
// PlivoOptions.cs
//
// (3.3.0) Plivo v1 API credentials + AnswerUrl base for media-stream
// XML. Empty AuthId/AuthToken → fail-soft.

using System;

namespace CircleAI.Telephony.Plivo;

/// <summary>(3.3.0) Plivo account credentials + endpoint.</summary>
public sealed class PlivoOptions
{
    /// <summary>Plivo v1 API base address. Default <c>https://api.plivo.com</c>.</summary>
    public Uri BaseAddress { get; init; } = new("https://api.plivo.com");

    /// <summary>Plivo Auth ID (starts with "MA..." or similar).</summary>
    public string? AuthId { get; init; }

    /// <summary>Plivo Auth Token.</summary>
    public string? AuthToken { get; init; }

    /// <summary>
    /// (Required for dial) HTTPS URL the host serves that, given a
    /// <c>?stream=&lt;url-encoded wss://...&gt;</c> query parameter, returns
    /// Plivo XML containing the matching <c>&lt;Stream/&gt;</c> verb.
    /// </summary>
    public Uri? AnswerUrlBase { get; init; }
}

File: python/src/circle_ai/telephony/plivo_options.py
# plivo_options.py
#
# Port of CircleAI.Telephony.Plivo/PlivoOptions.cs (C# — the EXACT spec).
#
# (3.3.0) Plivo v1 API credentials + AnswerUrl base for media-stream XML. Empty
# AuthId/AuthToken -> fail-soft.

from __future__ import annotations

from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True, slots=True)
class PlivoOptions:
    """(3.3.0) Plivo account credentials + endpoint. Mirrors ``PlivoOptions``."""

    #: Plivo v1 API base address. Default ``https://api.plivo.com``.
    base_address: str = "https://api.plivo.com"
    #: Plivo Auth ID (starts with "MA..." or similar).
    auth_id: Optional[str] = None
    #: Plivo Auth Token.
    auth_token: Optional[str] = None
    #: (Required for dial) HTTPS URL the host serves that, given a
    #: ``?stream=<url-encoded wss://...>`` query parameter, returns Plivo XML
    #: containing the matching ``<Stream/>`` verb.
    answer_url_base: Optional[str] = None


__all__ = ["PlivoOptions"]
